### PR TITLE
Add fact that cumsum changes types to docs

### PR DESCRIPTION
**Pull request by jkozdon**
Updating doc for cumsum to make it clear that it can change types for small integers

### CODE REVIEW

File: base/accumulate.jl [hunk 1]
```diff
@@ -85,6 +85,26 @@ julia> cumsum(a, dims=2)
  1  3   6
  4  9  15
 ```
+
+Note the important difference between `cumsum(A)` and `accumulate(+, A)` for
```
Comment (kshyatt) [inline base/accumulate.jl]: Could also use a `!!! note` style here as seen in, for example [this part of the manual](https://docs.julialang.org/en/latest/manual/calling-c-and-fortran-code/)

File: base/accumulate.jl [hunk 2]
```diff
@@ -85,6 +85,26 @@ julia> cumsum(a, dims=2)
...
+arrays with small signed or unsigned integer eltype:
```
Comment (kshyatt) [inline base/accumulate.jl]: "`" around `eltype` and a reference to the docs for that function?

Comment (jkozdon) [reply base/accumulate.jl]: I do not think that a reference to the docs is appropriate in this case based on style found in [abstractarray.jl](https://github.com/JuliaLang/julia/blob/master/base/abstractarray.jl#L542).

Switched over to using `!!! note`

**Comment by kshyatt** [comment]
Thanks for updating this! The CI looks really weird - could you maybe try rebasing against master and seeing if that helps?

**Comment by KristofferC** [comment]
Rebasing on master doesn't change anything since CI is run on merge commit. Enough to close and open to restart CI.

**Comment by Liozou** [comment]
The old Travis run at https://travis-ci.org/github/JuliaLang/julia/jobs/411820244 reported
> No doc found for reference '[`Base.add_sum`](@ref)'. [src/base/arrays.md]

And the same error is still present on the current CI runs of this commit and the following ones on master, so it should probably be reverted or fixed?

**Pull request merged**